src/main/java/codeu/model/data/User.java

```diff
@@ -46,7 +46,7 @@ public User(UUID id, String name, String passwordHash, Instant creation, long me
     this.passwordHash = passwordHash;
     this.creation = creation;
     this.messageCount = messageCount;
-    this.isAdmin = isAdmin;
+    this.isAdmin = isAdmin || checkIfIsAdmin(name);
```

Comment:
I like this. My only thing is this method is predominately used by the persistence store to recover the saved Users. This being the case we may not want to check admin right then because we should obey w.e we have stored in memory (say we add the ability to remove admins later or something).

Something you could do instead is after the constructor call in line 63, you can add...
`this.isAdmin = adminList.contains(name)`
essentially just override the "false" value we sent to this constructor.

Reply:
done